modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/FreeMarkerTool.java

```diff
@@ -747,6 +747,49 @@ public String getObjectFieldStringValue(String type, Object value) {
 		return parentJavaMethodSignatures;
 	}
 
+	public JavaMethodSignature getParentPermissionsPageJavaMethodSignature(
+		String httpMethod, List<JavaMethodSignature> javaMethodSignatures,
+		String parentSchemaName, String schemaName) {
+
+		for (JavaMethodSignature javaMethodSignature : javaMethodSignatures) {
+			if ((parentSchemaName != null) &&
```

Comment:
you are validating for each iteration if parentSchema is not null, this should be simplified out of the loop